src/entities/state_adapter.ts

```diff
@@ -0,0 +1,26 @@
+import createNextState, { Draft } from 'immer'
+import { EntityState } from './models'
+import { PayloadAction, isFSA } from '../createAction'
+
+export function createStateOperator<V, R>(
+  mutator: (arg: R, state: EntityState<V>) => void
+): EntityState<V>
+export function createStateOperator<V, R>(
+  mutator: (arg: any, state: any) => void
+): any {
+  return function operation<S extends EntityState<V>>(
+    state: any,
+    arg: R | PayloadAction<R>
+  ): S {
+    // @ts-ignore createNextState() produces an Immutable<Draft<S>> rather
+    // than an Immutable<S>, and TypeScript cannot find out how to reconcile
+    // these two types.
+    return createNextState(state, (draft: Draft<EntityState<V>>) => {
```

Comment:
If state is already an immer instance, would it make sense to keep the existing draft so it could be further modified in the reducer after any entity updates happen instead of returning a transformed object?

Reply:
Hmm.  Y'know, that may be a potential solution for the problem I'm seeing over in https://github.com/immerjs/immer/issues/377 .

It looks like there is an `isDraft` function exported by Immer:

https://immerjs.github.io/immer/docs/original

So yeah, I could very likely check that, and if it's a draft, just immediately call the `mutator` function instead of wrapping in `createNextState`.  Otherwise, assume this is being called standalone.

Reply:
Makes sense to me. That behavior would make the entities usage really straightforward.

Reply:
Yeah, that worked!  Published as [`alpha.6`](https://github.com/reduxjs/redux-toolkit/releases/tag/v1.3.0-alpha.6)